### PR TITLE
Support ES version 7.6.1

### PR DESCRIPTION
**Pull request by bowenlan-amzn**
Issue #, if available: #47 

*Description of changes:*
Support ES Version 7.6.1

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by bowenlan-amzn** [comment]
The JDK version in Test and Build Workflow need to be 13 for this version

File: sample-extension-plugin/build.gradle [hunk 1]
```diff
@@ -34,13 +35,7 @@ dependencies {
...
-    distribution = "oss-zip"
-    plugin(rootProject.getPath())
-}
-
-run {
-    distribution = "oss-zip"
-    plugin(rootProject.getPath())
+testClusters.integTest {
+    testDistribution = 'oss'
+    // plugin(rootProject.getPath())
```
Comment (dbbaughe) [inline sample-extension-plugin/build.gradle]: Is this not needed anymore?

File: build.gradle [hunk 2]
```diff
@@ -60,13 +61,13 @@ esplugin {
     classname 'com.amazon.opendistroforelasticsearch.jobscheduler.JobSchedulerPlugin'
 }
 
-integTestRunner {
-    // add "-Dtests.security.manager=false" to VM options if you want to run integ tests in IntelliJ
-    systemProperty 'tests.security.manager', 'false'
+testClusters.integTest {
+    testDistribution = 'oss'
```
Comment (dbbaughe) [inline build.gradle]: Capitalize to `OSS`

File: sample-extension-plugin/build.gradle [hunk 3]
```diff
@@ -34,13 +35,7 @@ dependencies {
...
-integTestCluster {
-    distribution = "oss-zip"
-    plugin(rootProject.getPath())
-}
-
-run {
-    distribution = "oss-zip"
-    plugin(rootProject.getPath())
+testClusters.integTest {
+    testDistribution = 'oss'
```
Comment (dbbaughe) [inline sample-extension-plugin/build.gradle]: Capitalize to `OSS`

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,20 +1,6 @@
...
-#       http://www.apache.org/licenses/LICENSE-2.0
-#
-#   or in the "license" file accompanying this file. This file is distributed
-#   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
-#   express or implied. See the License for the specific language governing
-#   permissions and limitations under the License.
-#
-
+#Tue Mar 17 23:27:16 PDT 2020
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.2.2-all.zip
```
Comment (dbbaughe) [inline gradle/wrapper/gradle-wrapper.properties:2]: When upgrading gradle versions we also need to check in the upgraded wrapper

**Review by dbbaughe — COMMENTED** [review]
Can you also upgrade the git workflow file that is causing the CI to fail

File: src/test/java/com/amazon/opendistroforelasticsearch/jobscheduler/JobSchedulerPluginTests.java [hunk 5]
```diff
@@ -16,11 +16,11 @@
 package com.amazon.opendistroforelasticsearch.jobscheduler;
 
 import org.junit.Assert;
-import org.junit.Test;
+// import org.junit.Test;
```
Comment (dbbaughe) [inline src/test/java/com/amazon/opendistroforelasticsearch/jobscheduler/JobSchedulerPluginTests.java]: Two things - What was reason for removing? And if you comment something out, just completely remove it, we can always refer to git history if we need to get something back, commented out code is just dead code that isn't used so don't be shy to get rid of it.

Comment (bowenlan-amzn) [reply src/test/java/com/amazon/opendistroforelasticsearch/jobscheduler/JobSchedulerPluginTests.java]: IDE shows me `@Test` annotation is not needed, a method named like `testFooBar` is enough. 
Guess comment but not remove is a bad habit of me, will correct that, thanks for point out.

File: sample-extension-plugin/build.gradle (hunk 1 above)
Comment (bowenlan-amzn) [reply sample-extension-plugin/build.gradle]: This line is to install job-scheduler before install sample plugin. Not sure if new ES test framework will install job-scheduler automatically.

Comment (dbbaughe) [reply sample-extension-plugin/build.gradle]: This breaks sample plugin.

```
> Task :opendistro-job-scheduler-sample-extension:run FAILED
Exec output and error:
| Output for ./bin/elasticsearch-plugin:-> Installing file:/Users/dbbaughe/opendistro/job-scheduler/sample-extension-plugin/build/distributions/opendistro-job-scheduler-sample-extension-1.6.0.0-SNAPSHOT.zip
| -> Downloading file:/Users/dbbaughe/opendistro/job-scheduler/sample-extension-plugin/build/distributions/opendistro-job-scheduler-sample-extension-1.6.0.0-SNAPSHOT.zip
| -> Failed installing file:/Users/dbbaughe/opendistro/job-scheduler/sample-extension-plugin/build/distributions/opendistro-job-scheduler-sample-extension-1.6.0.0-SNAPSHOT.zip
| -> Rolling back file:/Users/dbbaughe/opendistro/job-scheduler/sample-extension-plugin/build/distributions/opendistro-job-scheduler-sample-extension-1.6.0.0-SNAPSHOT.zip
| -> Rolled back file:/Users/dbbaughe/opendistro/job-scheduler/sample-extension-plugin/build/distributions/opendistro-job-scheduler-sample-extension-1.6.0.0-SNAPSHOT.zip
| Exception in thread "main" java.lang.IllegalArgumentException: Missing plugin [opendistro-job-scheduler], dependency of [opendistro-job-scheduler-sample-extension]
|       at org.elasticsearch.plugins.PluginsService.addSortedBundle(PluginsService.java:452)
|       at org.elasticsearch.plugins.PluginsService.sortBundles(PluginsService.java:424)
|       at org.elasticsearch.plugins.InstallPluginCommand.jarHellCheck(InstallPluginCommand.java:847)
|       at org.elasticsearch.plugins.InstallPluginCommand.loadPluginInfo(InstallPluginCommand.java:821)
|       at org.elasticsearch.plugins.InstallPluginCommand.installPlugin(InstallPluginCommand.java:866)
|       at org.elasticsearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:254)
|       at org.elasticsearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:224)
|       at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86)
|       at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:125)
|       at org.elasticsearch.cli.MultiCommand.execute(MultiCommand.java:91)
|       at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:125)
|       at org.elasticsearch.cli.Command.main(Command.java:90)
|       at org.elasticsearch.plugins.PluginCli.main(PluginCli.java:47)

FAILURE: Build failed with an exception.
```

Comment (bowenlan-amzn) [reply sample-extension-plugin/build.gradle]: I found old `plugin` method is in **clusterFormation** which is not used in our new ES test framework. The new `plugin` method in **testCluster** needs a plugin.zip file, so we will need to build or assemble job-scheduler first and then use the plugin.zip for sample plugin tasks.

**Review by dbbaughe — APPROVED** [review]

**Review by zengyan-amazon — APPROVED** [review]
let this change in for now

**Pull request merged**